samples/junit-tests/src/test/java/com/yelp/codegen/generatecodesamples/PostFileEndpointTest.kt

```diff
@@ -0,0 +1,179 @@
+package com.yelp.codegen.generatecodesamples
+
+import com.yelp.codegen.generatecodesamples.apis.FileApi
+import com.yelp.codegen.generatecodesamples.tools.MockServerApiRule
+import okhttp3.MediaType
+import okhttp3.RequestBody
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+
+private data class FileApiStats(
+    var boundary: String? = null,
+    var errors: List<String>? = null,
+    var parts: List<MultiPartInfo>? = null
+)
+private data class MultiPartInfo(
+    val content: String? = null,
+    val contentDisposition: String? = null,
+    val contentLength: String? = null,
+    val contentType: String? = null
+)
+
+private fun processPart(part: String): MultiPartInfo {
+
+    val partLines = part.split("\r\n").let {
+        // part would look like "\r\nHeader-Name: Header-Value\r\n...\r\n\r\nbody\r\n"
+        // Here we're removing the first and last part as we know that are empty and
+        it.subList(1, it.size - 1)
+    }
+
+    val partLinesIterator = partLines.iterator()
+    val headers = mutableMapOf<String, String>()
```

Comment:
```kotlin
val headers = part.split("\r\n")
            .drop(1)
            .dropLast(1)
            .filter { it.isNotEmpty() }
            .map { it.split(": ", limit = 2) }
            .associateBy({it[0]}, {it[1]})
```

Reply:
I was using the iterator to traverse the all input only one.
It's kinda sad that kotlin (at least for what I found) does not allow to scan an iterator while consuming it

The subsequent variable `content` represent the content of the file.

Considering this I would prefer using the _iterator_ approach at it makes simpler to iterate to all the lines (`part.split("/r/n")`) and deal with the two groups in a different way

Reply:
> I was using the iterator to traverse the all input only one.

I mean, it's inside a test 😅 readability >> perfomance.
I'd rather go for the functional approach rather than using an iterator over here.

Reply:
Also this could work for the content:

```
val content  = headers.entries.fold("") { previous, current -> "$previous\r\n${current.key}: ${current.value}" }
```

Reply:
I see your point, I'm going to use a more functional style to implement this function.

I think that we have some misunderstanding here, so I'll write down an example to be more explicative.

part will look like
```
--BOUNDARY\r\n
header1: value1\r\n
header2: value2\r\n
\r\n
fileContent\r\n

```
and we want to extract:
 * `mapOf("header1" to "value1", "header2" to "value2")`  into `headers`
 * `fileContent` into `content`